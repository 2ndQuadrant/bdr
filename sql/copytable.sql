\c postgres

CREATE TABLE testtab (x integer, blah text);

INSERT INTO testtab(x, blah) SELECT x, repeat('blah'||x, x) FROM generate_series(1,10) x;

SELECT pg_xlog_wait_remote_apply(pg_current_xlog_location(), pid) FROM pg_stat_replication;

\c regression

CREATE TABLE testtab (x integer, blah text);

CREATE FUNCTION bdr_copytable_test(cstring,cstring,cstring,cstring) RETURNS void LANGUAGE c AS 'bdr.so','bdr_copytable_test';

SELECT bdr_copytable_test(
	'dbname=postgres',
	'dbname=regression',
	'COPY public.testtab TO stdout',
	'COPY public.testtab FROM stdin');

-- Shouldn't be needed. Wha?
SELECT pg_sleep(1);

SELECT * FROM testtab ORDER BY x;

DROP FUNCTION bdr_copytable_test(cstring,cstring,cstring,cstring);

SELECT pg_xlog_wait_remote_apply(pg_current_xlog_location(), pid) FROM pg_stat_replication;

\c postgres

SELECT * FROM testtab ORDER BY x;
