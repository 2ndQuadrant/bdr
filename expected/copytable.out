\c postgres
CREATE TABLE testtab (x integer, blah text);
INSERT INTO testtab(x, blah) SELECT x, repeat('blah'||x, x) FROM generate_series(1,10) x;
SELECT pg_xlog_wait_remote_apply(pg_current_xlog_location(), pid) FROM pg_stat_replication;
 pg_xlog_wait_remote_apply 
---------------------------
 
 
(2 rows)

\c regression
CREATE TABLE testtab (x integer, blah text);
ERROR:  relation "testtab" already exists
CREATE FUNCTION bdr_copytable_test(cstring,cstring,cstring,cstring) RETURNS void LANGUAGE c AS 'bdr.so','bdr_copytable_test';
SELECT bdr_copytable_test(
	'dbname=postgres',
	'dbname=regression',
	'COPY public.testtab TO stdout',
	'COPY public.testtab FROM stdin');
 bdr_copytable_test 
--------------------
 
(1 row)

-- Shouldn't be needed. Wha?
SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

SELECT * FROM testtab ORDER BY x;
 x  |                             blah                             
----+--------------------------------------------------------------
  1 | blah1
  1 | blah1
  2 | blah2blah2
  2 | blah2blah2
  3 | blah3blah3blah3
  3 | blah3blah3blah3
  4 | blah4blah4blah4blah4
  4 | blah4blah4blah4blah4
  5 | blah5blah5blah5blah5blah5
  5 | blah5blah5blah5blah5blah5
  6 | blah6blah6blah6blah6blah6blah6
  6 | blah6blah6blah6blah6blah6blah6
  7 | blah7blah7blah7blah7blah7blah7blah7
  7 | blah7blah7blah7blah7blah7blah7blah7
  8 | blah8blah8blah8blah8blah8blah8blah8blah8
  8 | blah8blah8blah8blah8blah8blah8blah8blah8
  9 | blah9blah9blah9blah9blah9blah9blah9blah9blah9
  9 | blah9blah9blah9blah9blah9blah9blah9blah9blah9
 10 | blah10blah10blah10blah10blah10blah10blah10blah10blah10blah10
 10 | blah10blah10blah10blah10blah10blah10blah10blah10blah10blah10
(20 rows)

DROP FUNCTION bdr_copytable_test(cstring,cstring,cstring,cstring);
SELECT pg_xlog_wait_remote_apply(pg_current_xlog_location(), pid) FROM pg_stat_replication;
 pg_xlog_wait_remote_apply 
---------------------------
 
 
(2 rows)

\c postgres
SELECT * FROM testtab ORDER BY x;
 x  |                             blah                             
----+--------------------------------------------------------------
  1 | blah1
  1 | blah1
  2 | blah2blah2
  2 | blah2blah2
  3 | blah3blah3blah3
  3 | blah3blah3blah3
  4 | blah4blah4blah4blah4
  4 | blah4blah4blah4blah4
  5 | blah5blah5blah5blah5blah5
  5 | blah5blah5blah5blah5blah5
  6 | blah6blah6blah6blah6blah6blah6
  6 | blah6blah6blah6blah6blah6blah6
  7 | blah7blah7blah7blah7blah7blah7blah7
  7 | blah7blah7blah7blah7blah7blah7blah7
  8 | blah8blah8blah8blah8blah8blah8blah8blah8
  8 | blah8blah8blah8blah8blah8blah8blah8blah8
  9 | blah9blah9blah9blah9blah9blah9blah9blah9blah9
  9 | blah9blah9blah9blah9blah9blah9blah9blah9blah9
 10 | blah10blah10blah10blah10blah10blah10blah10blah10blah10blah10
 10 | blah10blah10blah10blah10blah10blah10blah10blah10blah10blah10
(20 rows)

